igvm/hypervisor.py

```diff
@@ -154,6 +155,18 @@ def check_vm(self, vm):
                 .format(free_mib, vm.dataset_obj['memory'])
             )
 
+        # Ensure that online migration is possible for given os
+        if not offline:
+            if ((vm.dataset_obj['os'], self.dataset_obj['os'])
+                    not in MIGRATE_COMMANDS
+               ):
+                raise HypervisorError(
+                    '{} to {} migration is not supported.'
+                    .format(
+                        vm.dataset_obj['os'], self.dataset_obj['os']
+                    )
+                )
```

Comment:
I disagree. (But I might be wrong here).
It makes sense to check before starting a migration if the source and target os are compatible.
One person could specify a hypervisor to migrate to but it could have been a bad choice. In this case we want to fail here.

We also need this check when trying to find a best hypervisor as we don't want to select a source=wheezy and destination=stretch combination.